### DISCUSSION TITLE
Importing scalar image and its binary labelmap in NIFTI format

### DISCUSSION THREAD

**Discussion by AbdulrahmanMohammed2021** [TorchIO-project/torchio#483] [General]
Hi Fernando,
I've imported a MRI scan of head and its binary labelmap as a volume and as segmentation, respectively. However, the output volume didn't align with segmentation. So can you please give any explanation for that?

Note: I'm wanna to save the augmented volume and its segmentation(binary labelmap) as actual augmentation in NIFTI format. I'm using MATLAB so that the reason why I can't perform augmentation during training.

Thanks

![TorchIO](https://user-images.githubusercontent.com/80439295/110734681-6fe89080-827c-11eb-9a17-ea3914f596ca.png)

**Comment by fepegar** [comment]
Hi, @AbdulrahmanMohammed2021. This is not related to TorchIO. You probably have different spatial metadata for both images.

**Reply by AbdulrahmanMohammed2021** [reply]
Hi,@Fernando, just to let you know that I've done the segmentation on Slicer, and exported images and their segmentation into NIFTI files. Then in another Slicer scene, I imported them ( scalar image as a volume and volume labelmap  as segmentation) to perform augmentation using TorchIO.  When I applied the augmentation, it is only applied to the scalar image and It doesn't align with the labelmap as shown in the previous post. Can you elaborate on that?

I hope it is clear

thanks

**Reply by fepegar** [reply]
Thanks for clarifying. Although TorchIO performs the same operation to all images in the `Subject`, the Slicer module doesn't have support for that. That is because it is conceived to play with the transforms parameters, not to perform the actual augmentation for training. If you use the snippet I posted on the Slicer forum, your transformed images will be correctly aligned.

**Reply by AbdulrahmanMohammed2021** [reply]
OK, that's make sense I've tried the snippet and as you said, they were aligned correctly. last question, can I save the transformed images in NIFTI format?

thanks

**Reply by fepegar** [reply]
Sure! You can use the [`save()`](https://torchio.readthedocs.io/data/image.html) method of the images.
